grisera_api/registered_channel/registered_channel_service_mongodb.py

```diff
@@ -0,0 +1,191 @@
+
+        return self.get_registered_channel(registered_channel_id)
+
+    def get_registered_channels(self):
+        """
+        Send request to mongo api to get registered channels
+
+        Returns:
+            Result of request as list of registered channels objects
+        """
+        registered_channels = mongo_api_service.db.registered_channels.find({})
+        result = [RegisteredChannelIn(**rc) for rc in registered_channels]
+
+        return result
+
+    def get_registered_channel(
+        self, registered_channel_id: int, depth: int = 0, source: str = ""
+    ):
+        """
+        Send request to graph api to get given registered channel
+
+        Args:
+            registered_channel_id (int): Id of registered channel
+
+        Returns:
+            Result of request as registered channel object
+        """
+        registered_channel = mongo_api_service.db.registered_channels.find_one(
+            {"id": registered_channel_id}
+        )
+
+        if registered_channel is None:
+            return NotFoundByIdModel(
+                id=registered_channel_id,
+                errors={"errors": "registered channel not found"},
+            )
+
+        if depth == 0:
+            return RegisteredChannelOut(**registered_channel)
+
+        if source != "recording":
+            registered_channel["recorgings"] = self.recording_service.get_recordings(
+                {"registered_channel": registered_channel["id"]},
+                depth=depth - 1,
+                source="registered_channel",
+            )
+
+        has_related_channel = registered_channel["channel_id"] is not None
+        if source != "channel" and has_related_channel:
+            registered_channel["channel"] = self.channel_service.get_channel(
+                channel_id=registered_channel["channel_id"],
+                depth=depth - 1,
+                source="registered_channel",
+            )
+
+        has_related_rd = registered_channel["registered_data_id"] is not None
+        if source != "registered_data" and has_related_rd:
+            registered_channel[
+                "registered_data"
+            ] = self.channel_service.get_registered_data(
+                registered_data_id=registered_channel["registered_data_id"],
+                depth=depth - 1,
+                source="registered_channel",
+            )
+
+        return RegisteredChannelOut(**registered_channel)
+
+    def delete_registered_channel(self, registered_channel_id: int):
+        """
+        Send request to graph api to delete given registered channel
+
+        Args:
+            registered_channel_id (int): Id of registered channel
+
+        Returns:
+            Result of request as registered channel object
+        """
+        get_response = self.get_registered_channel(registered_channel_id)
+
+        if type(get_response) is NotFoundByIdModel:
+            return get_response
+
+        self.graph_api_service.delete_node(registered_channel_id)
```

Comment:
variable `graph_api_service` not present in class

Reply:
I have deleted the file. This file was added for presentation purpose only.